libportal/inputcapture-pointerbarrier.c

```diff
@@ -0,0 +1,255 @@
+/*
+ * Copyright (C) 2022, Red Hat, Inc.
+ *
+ * This file is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation, version 3.0 of the
+ * License.
+ *
+ * This file is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * SPDX-License-Identifier: LGPL-3.0-only
+ */
+
+#include "config.h"
+
+#include "portal-private.h"
+#include "session-private.h"
+#include "inputcapture-pointerbarrier.h"
+#include "inputcapture-private.h"
+
+/**
+ * XdpInputCapturePointerBarrier
+ *
+ * A representation of a pointer barrier on an [class@InputCaptureZone].
+ * Barriers can be assigned with
+ * [method@InputCaptureSession.set_pointer_barriers], once the Portal
+ * interaction is complete the barrier's "is-active" state indicates whether
+ * the barrier is active. Barriers can only be used once, subsequent calls to
+ * [method@InputCaptureSession.set_pointer_barriers] will invalidate all
+ * current barriers.
+ */
+
+enum
+{
+  PROP_0,
+
+  PROP_X1,
+  PROP_X2,
+  PROP_Y1,
+  PROP_Y2,
+  PROP_ID,
+  PROP_IS_ACTIVE,
+
+  N_PROPERTIES
+};
+
+enum
+{
+  LAST_SIGNAL
+};
+
+enum barrier_state
+{
+  BARRIER_STATE_NEW,
+  BARRIER_STATE_ACTIVE,
+  BARRIER_STATE_FAILED,
+};
+
+static GParamSpec *properties[N_PROPERTIES] = { NULL, };
+
+struct  _XdpInputCapturePointerBarrier {
+    GObject parent_instance;
+
+    guint id;
+    gint x1, y1;
+    gint x2, y2;
+
+    enum barrier_state state;
+};
+
+G_DEFINE_TYPE (XdpInputCapturePointerBarrier, xdp_input_capture_pointer_barrier, G_TYPE_OBJECT)
+
+static void
+xdp_input_capture_pointer_barrier_finalize (GObject *object)
+{
+  G_OBJECT_CLASS (xdp_input_capture_pointer_barrier_parent_class)->finalize (object);
+}
```

Comment:
If the implementation is empty, there's no need to override.

Reply:
Removed as well.